python/tank/template.py

```diff
@@ -703,12 +703,26 @@ def make_template_paths(data, keys, all_per_platform_roots):
 
     :returns: Dictionary of form {<template name> : <TemplatePath object>}
     """
+
+    if data and not all_per_platform_roots:
+        raise TankError(
+            "At least one root must be defined when using 'path' templates."
+        )
+
     template_paths = {}
     templates_data = _process_templates_data(data, "path")
 
     for template_name, template_data in templates_data.items():
         definition = template_data["definition"]
-        root_name = template_data["root_name"]
+        root_name = template_data.get("root_name")
```

Comment:
is this the bit which let's us omit root_name from template.yml? I think we should hold off doing that for the time being - it's not strictly necessary in order to allow studios to rename 'primary' for their single-root configs, right?

Reply:
We need to keep it, otherwise any existing config which does not have an explicit `root_name` will break. Current tk-core supports this but use "primary" as the default.

Reply:
gotcha!